src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -506,8 +506,25 @@ int32_t get_default_ddr_index(int32_t dev_index, int32_t cu_index) {
 }
 
 void xma_enable_mode1(void) {
+    if (xrt_core::config::get_xma_mode2()) {
+        xma_logmsg(XMA_ERROR_LOG, XMAUTILS_MOD, "xma_enable_mode1: Mode2 is selected. Use either mode1 or mode2");
+        return;
+    }
+    if (g_xma_singleton->num_execbos != 3) {
+        xma_logmsg(XMA_ERROR_LOG, XMAUTILS_MOD, "xma_enable_mode1: Mode already changed");
```

Comment:
Update the error message:
xma_enable_mode1: Mode1 already in effect, no action taken.

Reply:
done